EIPS/eip-remove_utf8_support.md

```diff
@@ -0,0 +1,140 @@
+
+The new string header would instead have three parts: *length*, *range*, and *reserved*.
+
+| Section              | position (little endian)  |  size  | total size |
+|----------------------|---------------------------|--------|------------|
+| length (bytes)       | 0                         | 8      | 8          |
+| range                | 8                         | 8      | 16         |
+| reserved             | 16                        | 16     | 32         |
+
+##### length
+
+The length is the length of the data in bytes (same as it is now).
+
+##### range
+
+The range is an unsigned 8 byte integer, which is to be interpreted as the upper bound for character values. The lower bound is always `0`.
+
+A range of `0` means that the standard byte encoding is used, so it is the same as setting it to `255`. This is to make `string` compatible with `bytes`.
+
+Validating a string is done by checking that none of its characters has a value that is outside of the range.
+
+Note that non power-of-two character-widths are not ideal, because they lead to inefficient use of storage (a single value could have its bytes spread out over more then one storage slot).
+
+##### reserved
+
+Reserved space can be used to store additional information. Using it is optional, but it can not be assumed to be empty.
+
+#### Using the reserved section
+
+The reserved section could be used to make more advanced strings possible.
+
+| Section              | position (little endian)  |  size  | total size |
+|----------------------|---------------------------|--------|------------|
+| length (bytes)       | 0                         | 8      | 8          |
+| range                | 8                         | 8      | 16         |
+| format               | 16                        | 8      | 24         |
+| reserved             | 24                        | 7      | 31         |
+| extra bytes          | 31                        | 1      | 32         |
+
+##### format
+
+The format is an identifier, and would usually indicate that the string has a particular format. This format could for example be backed by a *character mapping*, but that is not a requirement (see the character mappings section for more information).
+
+The zero key (i.e. `key == 0`) would be reserved for the standard byte encoding (range `0`).
+
+##### extra bytes (optional)
+
+The extra bytes field is optional, but can (and should) be used to avoid having to recompute the character width every time it is needed. The value held in this field must be one less then the (smallest) amount of bytes that is required to hold a character. Putting it in the last byte means it can be accessed using a single bit-shift operation (no masking is needed), which means that index access and length computation becomes slightly more efficient.
+
+The reason for using extra bytes and not the character width itself is to stay compatible with the standard byte encoding.
```

Comment:
What does `extra bytes = 8` mean, for example?  Is it somehow related to character widths?

Reply:
yeah, it just means the exact number of bytes required for each character minus one.

it was just a practical thing. i noticed that having to compute the number of bytes required to store a number of range `X` every time you do something with a string was more annoying and time consuming then just pre-computing and keeping it in the header. 

the reason it's "extra bytes" instead of just "number of bytes" is to avoid putting a number there if the bytes required for the range is 1. With the convention of interpreting a range of `0` as the standard byte encoding (having the actual range of `255`), then extra bytes is `0` too and any normal byte string would automatically be compatible with this format. At least if its length value is less then 8 bytes in size...